netevents/listener.go

```diff
@@ -0,0 +1,81 @@
+package netevents
+
+import (
+	"net"
+	"sync/atomic"
+
+	"github.com/segmentio/events"
+	"github.com/segmentio/netx"
+)
+
+// NewListener returns a wrapped version of lstn which logs with the default
+// logger all accept and close events that occur on the listener.
+func NewListener(lstn net.Listener) net.Listener {
+	return NewListenerWith(events.DefaultLogger, lstn)
+}
+
+// NewListenerWith returns a wrapped version of lstn which logs with logger
+// all accept and close events that occur on the listener.
+func NewListenerWith(logger *events.Logger, lstn net.Listener) net.Listener {
+	return &listener{
+		lstn:   lstn,
+		logger: logger,
+	}
+}
+
+type listener struct {
+	lstn   net.Listener
+	logger *events.Logger
+	closed uint32
+}
+
+func (l *listener) Accept() (conn net.Conn, err error) {
+	if conn, err = l.lstn.Accept(); err != nil {
+		if atomic.LoadUint32(&l.closed) == 0 {
+			l.error("accept", err)
+		}
+	}
+
+	if conn != nil {
```

Comment:
I doubt it, but I'd rather not make assumptions about how the base `net.Listener` is implemented ;)